torch/csrc/jit/codegen/cuda/ops/alias.h

```diff
@@ -39,6 +39,24 @@ TORCH_CUDA_CU_API TensorView* squeeze(
 
 TORCH_CUDA_CU_API TensorView* unsqueeze(TensorView* x, int dim);
 
+//! Permute a tensor as specified by axis mappings.
+//!
+//! The transposition mapping is specified with a list of pairs from
+//! new to old positions. Positions are relative to the noReduction
+//! domain.
+//!
+//! \param inp Tensor to transpose
+//! \param new2old vector mapping from new to old positions.
+TORCH_CUDA_CU_API TensorView* permute(
+    TensorView* x,
+    const std::vector<int64_t>& new2old);
+
+//! Transpose a tensor by swapping the two dimensions.
+TORCH_CUDA_CU_API TensorView* transpose(
+    TensorView* x,
+    int64_t dim0 = 0,
+    int64_t dim1 = 1);
```

Comment:
Do we always want to have these default values, .e.g., for higher dimensional tensors than 2D?

Reply:
Eager mode only has default values for the 2D case. https://pytorch.org/docs/stable/generated/torch.t.html
We could have the default values be `dim0 = -1` and `dim1 = -2`. I'm also fine removing the default values.

Reply:
Why don't we create an overload of `transpose` with no axes parameters. That would do `TORCH_CHECK` to make sure the input tensor is 2D, and then it would call the other version of `transpose` with `dim0 = 0` and `dim1= 1`. 

Function named just as `t` might be fine too instead of adding a overload of `transpose`. Maybe that's better aligned with the eager mode?

Reply:
An overload of `transpose` make sense. Calling `t(tv0)` doesn't provide much context, so I don't like that option.